### ISSUE TITLE
Features needed for CMSSW_10_1_1

### ISSUE THREAD

**Issue by fabiocos** [cms-sw/cmssw#22865]
The validation of CMSSW_10_1_0 is ongoing, we already know that we need to back-port some fixes and some last minute planned additions. The target is a CMSSW_10_1_1 containing features already deployed and to be validated in 10_2_0_pre1 (the build and validation of these two releases will have to move unfortunately in parallel).

(Updated on April 7 23:00)

Here is the list of the PRs already backported from 10_2_X as fixes:

#22822 from cms-tsg-storm: Fix l1 emulation in HLT scripts (101X)
#22823 from folguera: BugFix for L3MuonCandidateProducerFromMuons
#22819 from bsunanda: Run2-alca126 Add eta dependent threshold for ECAL
#22818 from mmusich: [10.1.X] fix segmentation fault in SiStripBadComponentInfo (backport)
#22792 from thomreis: Fix L1T muon copy comparison DQM harvesting path - 101x
#22791 from mmusich: [10.1.X] remove useless triplication of AAG data in calibTree
#22800 from hroskes: fix preexisting tracker alignment validation

This issue is meant to collect the other  features that are really necessary for the operations at the beginning of data taking and for the completion of the PF calibration.

The list that I am aware of at present is:

- [x] #22827 Fix the product name for the PCL BeamSpot HP workflow (already in master)
- [ ] #22831 O2O scripts (TO BE SIGNED, already in master, on hold pending clarification by @ggovi)
- [ ] #22832 Fix for the popcon procedure (TO BE SIGNED, already in master)
- [x] #22897 GT in autoCond for 10_1_1, combination of PR 22834 and 22893
- [ ] #22844 DQM: SiPixelPhase1- Added Big Pixel Charge Plots [Backport for 10_1_X of #22324 ] (TO BE SIGNED, already in master)
- [x] #22858 CTPPS: updated DQM for 2018 data taking (10_1_X) (TO BE SIGNED)
- [x] #22866 ctpps dqm update - backport
- [x] #22860 Do not throw exception for fitting error for CMSSW_10_1
- [x] #22857 (sunanda alcareco fix)
- [ ] #22854 (SiPixelQuality memory cost mitigation)
- [ ] #22870 (for Lumi PCC PCL workflow, TO BE SIGNED)
- [x] #22864 (bug fix in EgammaHLTGsfTrackVarProducer - backport)
- [x] #22824 (backport for HLT menu migration to 10_1 template - already in 10_2_X)
- [x] #22794 Remove comparisons from GlobalAlgBlk equal operator - 101x
- [x] #22793 L1T DQM Ignore BX range mismatch in OMTF to uGMT comparison - 101x
- [x] #22795 L1T DQM FEDIntegrity update to stage2 - 101x

Please add further motivated requests to this issue

**Comment by Dr15Jones** [comment]
@fabiocos #22860 is needed to avoid the exception seen in AlcaBeamMonitor

**Comment by fabiocos** [comment]
@popovvp @fabferro  I understand that, besides @22858, for CTPPS we need also the backport of #22678 could you please provide it ?

**Comment by fabferro** [comment]
@fabiocos: yes we are going to open the backport of #22678 asap.

**Comment by lpernie** [comment]
From Alca side we need:
#22834  (2018 GT change)
#22857  (sunanda alcareco fix)
#22854  (SiPixelQuality memry cost mitigation)
#22858 and the backport of #22678 (ctpps fixes)
#22831 , #22832  (fix in O2O)
@ggovi  , @arunhep you can cross-check this list.

**Comment by fabferro** [comment]
I've opened PR #22866 as a backport of PR #22678 
@fabiocos @popovvp

**Comment by arunhep** [comment]
adding to what @lpernie already mentioned : 
https://github.com/cms-sw/cmssw/pull/22870 (for Lumi PCC PCL workflow)
and we are expecting a PR tomorrow which is needed to deploy new LHCInfo O2O
we will mention it here once it will be opened.

**Comment by Martin-Grunewald** [comment]
For TSG we request:

10_2_X/10_2_0_pre1
  #22863 (bug fix in EgammaHLTGsfTrackVarProducer)

10_1_X/10_1_1
  #22864 (bug fix in EgammaHLTGsfTrackVarProducer - backport)
  #22824 (backport for HLT menu migration to 10_1 template - already in 10_2_X)

**Comment by thomreis** [comment]
Hi @fabiocos 
For L1T we would like to have these in 10_1_1 if possible.
#22794 
#22793 
#22795 
#22880 
@rekovic please take a look as well.

**Comment by fabiocos** [comment]
@deguio as we discussed at the operation meeting, #22849 can come at a later stage without rushing (not yet approved in master), so I will remove it from the list

**Comment by fabiocos** [comment]
@arunhep @lpernie Huston we have a problem...

After the integration in 10_2_X of the GT PR #22833 (whose backport #22834 should be one of the key residual ingredients for 10_1_1) we are getting several failures in tonight's IB, see for instance 

https://cmssdt.cern.ch/SDT/cgi-bin/buildlogs/slc6_amd64_gcc630/CMSSW_10_2_X_2018-04-06-2300/pyRelValMatrixLogs/run/136.77_RunDoubleEG2016H+RunDoubleEG2016H+HLTDR2_2016+RECODR2_2016reHLT_skimDoubleEG_Prompt+HARVESTDR2/step3_RunDoubleEG2016H+RunDoubleEG2016H+HLTDR2_2016+RECODR2_2016reHLT_skimDoubleEG_Prompt+HARVESTDR2.log

They are reproducible, do not appear if only the GT PR is not used (In any case the other additions tonight look unrelated to this), and have the pattern

%MSG
----- Begin Fatal Exception 07-Apr-2018 07:13:54 CEST-----------------------
An exception of category 'Configuration' occurred while
   [0] Processing  Event run: 283877 lumi: 3 event: 3979066 stream: 2
   [1] Running path 'pathALCARECOEcalESAlign'
   [2] Calling method for module HLTHighLevel/'ALCARECOEcalESAlignHLT'
Exception Message:
 HLTHighLevel [instance: ALCARECOEcalESAlignHLT - path: pathALCARECOEcalESAlign]: No triggerList with key EcalESAlign in AlCaRecoTriggerBitsRcd
----- End Fatal Exception -------------------------------------------------

A test on some of the failing workflows is running in #22834 to cross check that this appears also in the 10_1_X branch. 

Any clue of what is wrong/missing? Needless to say that this is a show-stopper for 10_2_0_pre1 and 10_1_1 if not solved.

**Comment by arunhep** [comment]
@fabiocos i know the problem. i will open a PR soon.

**Comment by arunhep** [comment]
These PRs :
master : https://github.com/cms-sw/cmssw/pull/22892
10.1.X : https://github.com/cms-sw/cmssw/pull/22893
should fix the issue.

**Comment by fabiocos** [comment]
@arunhep #22897 combines #22834 and the fix #22893 (which should be rebased)

**Comment by fabiocos** [comment]
@fabozzi the new datasets look all at CERN, so I expect we should have no failures due to missing inputs

**Comment by fabiocos** [comment]
At this point I will freeze the present situation, more additions may come in a following patch/release
